### PR TITLE
fix: project structure

### PR DESCRIPTION
**Pull request by BondPV**
1. fix the project structure
2. fix settings webpack devServer
3. fix scss style header

### CODE REVIEW

**Review by Alesia-V175 — COMMENTED** [review]
Let's delete unnecessary rules in webpack.config file which deal with music and video

**Review by Alesia-V175 — APPROVED** [review]

File: src/components/app.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Catalog from './main/catalog/Catalog';
+
+class App {
```
Comment (Ksarelto) [inline src/components/app.ts]: App is a component, so he should have it's own folder App, and folders name === Component name

File: src/components/footer/footer.scss [hunk 2]
```diff
@@ -0,0 +1,35 @@
+.footer {
+  background-color: $bg-color-black;
+  padding: 10px 0;
+  &__wrap {
```
Comment (Ksarelto) [inline src/components/footer/footer.scss:5]: add empty line before this style block

File: src/components/footer/footer.scss [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  &__logo {
+    height: auto;
+    width: 80px;
+    opacity: 0.4;
+    
+    &:hover {
+      opacity: 1;
+      transition: 0.3s all;
```
Comment (Ksarelto) [inline src/components/footer/footer.scss]: if u want to use transition only when your mouse hover the logo ok, but if u want transition worked after you move out mouse from element u should add transition to element, not to pseudoclass

File: src/components/footer/footer.scss [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  &__github {
+    color: #ffffff;
+    text-decoration: none;
+    opacity: 0.5;
+
+    &:hover {
+      opacity: 1;
+      transition: 0.3s all;
```
Comment (Ksarelto) [inline src/components/footer/footer.scss]: same as above

File: src/components/header/contacts.scss [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  &__link {
+    color: $color-white;
+    padding-left: 5px;
+    transition: 0.3s;
+    text-decoration: none;
+    padding-right: 3px;
+
+    &:hover {
+      opacity: 0.6;
+      transition: 0.3s;
```
Comment (Ksarelto) [inline src/components/header/contacts.scss]: same vith transition, look at this in all scss

File: src/components/header/contacts.scss [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  &__messenger-icon {
+    display: block;
+    transition: 0.3s;
+
+    &:hover {
+      transform: scale(1.15);
+      opacity: 0.6;
+      transition: 0.3s;
```
Comment (Ksarelto) [inline src/components/header/contacts.scss]: here it is redundant, becuase u already added this value to element

File: src/components/header/header.scss [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: 36px;
+    margin: 0 8px;
+    transition: 0.3s;
+
+    &:hover {
+      transform: scale(1.15);
+      transition: 0.3s;
```
Comment (Ksarelto) [inline src/components/header/header.scss:55]: redundant style

File: src/components/main/catalog/product/Product.ts [hunk 8]
```diff
@@ -1,7 +1,7 @@
-import './productItem.scss';
-import { IProduct } from '../../types/types';
+import './product.scss';
+import { IProduct } from '../../../../types/interfaces';
```
Comment (Ksarelto) [inline src/components/main/catalog/product/Product.ts]: use aliase in webpack config, or this line of dotts will become very big in nested components

File: src/database/ProductsDB.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import productJSON from './database.json';
+import { IProducts } from '../types/interfaces';
+
+class ProductsDB {
+  jsonDB: IProducts;
+
+  constructor(jsonDB: IProducts) {
```
Comment (Ksarelto) [inline src/database/ProductsDB.ts]: u can create private var in constructor like constructor(private jsonDB: IProducts)

File: src/database/ProductsDB.ts [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    this.jsonDB = jsonDB;
+  }
+
+  getProducts() {
```
Comment (Ksarelto) [inline src/database/ProductsDB.ts]: u can set this methods as static, and dont create an instance of this class, just export ProductsDb class

File: src/components/footer/footer.scss (hunk 3 above)
Comment (BondPV) [reply src/components/footer/footer.scss]: сorrected

File: src/components/header/contacts.scss (hunk 6 above)
Comment (BondPV) [reply src/components/header/contacts.scss]: corrected

File: src/components/header/header.scss (hunk 7 above)
Comment (BondPV) [reply src/components/header/header.scss:55]: corrected

File: src/database/ProductsDB.ts (hunk 9 above)
Comment (BondPV) [reply src/database/ProductsDB.ts]: corrected

File: src/components/main/catalog/product/Product.ts (hunk 8 above)
Comment (BondPV) [reply src/components/main/catalog/product/Product.ts]: added aliases

File: src/components/app.ts (hunk 1 above)
Comment (BondPV) [reply src/components/app.ts]: added folder

File: src/components/footer/footer.scss (hunk 2 above)
Comment (BondPV) [reply src/components/footer/footer.scss:5]: added empty line

File: src/components/footer/footer.scss (hunk 4 above)
Comment (BondPV) [reply src/components/footer/footer.scss]: сorrected

File: src/components/header/contacts.scss (hunk 5 above)
Comment (BondPV) [reply src/components/header/contacts.scss]: сorrected

File: src/database/ProductsDB.ts (hunk 10 above)
Comment (BondPV) [reply src/database/ProductsDB.ts]: сorrected

**Pull request merged**